src/app/_components/HeadingWithIcon.tsx

```diff
@@ -0,0 +1,20 @@
+import { Heading } from '@/components/Heading'
+
+export function HeadingWithIcon({
+  icon,
+  children,
+}: {
+  icon: React.ReactNode
+  children: React.ReactNode
+}) {
+  return (
+    <div className="inline-flex items-center gap-3">
+      <span className="text-brand-300" aria-hidden={true}>
+        {icon}
+      </span>
+      <Heading tag="h3" variant="lg">
```

Comment:
At the moment, this would limit us to `h3` and `lg` at all times. This was arguably possible within `HomeExploreSectionCard` but now that we're abstracting it, we have a chance to change it.

Reply:
Since it's being used only in cards for now, would it be better to limit it for consistency? Or we could give it, for example, 3 sizes as have that as a prop? It would be more explicit.

Reply:
Also Heading component would need modifications as well as it has required props.